src/realm/set.cpp

```diff
@@ -320,4 +320,47 @@ void set_sorted_indices(size_t sz, std::vector<size_t>& indices, bool ascending)
     }
 }
 
+template <typename Iterator>
+static bool partition_points(const Set<Mixed>& set, std::vector<size_t>& indices, Iterator& first_string,
+                             Iterator& first_binary, Iterator& end)
+{
+    first_string = std::partition_point(indices.begin(), indices.end(), [&](size_t i) {
+        return set.get(i).is_type(type_Bool, type_Int, type_Float, type_Double, type_Decimal);
```

Comment:
A null value is a special case that is stored at the beginning of the list, I think it might need some special handling here to be included in the first partition. If it is working correctly, can you confirm with a test?

Reply:
The object store tests for Set include null in the values being tested and verify that std::sort and set.sort() give the same result. Null is at the start for both Set and regular sorting, as `SetElementEquals<Mixed> ` doesn't override the result for it.